OpenOversight/tests/test_utils.py

```diff
@@ -305,3 +309,29 @@ def test_crop_image_calls_upload_image_to_s3_and_store_in_db_with_user_id(
             assert (
                 current_user.get_id() in upload_image_to_s3_and_store_in_db.call_args[0]
             )
+
+
+@pytest.mark.parametrize(
+    "units, has_officers_with_unit, has_officers_with_no_unit",
+    [
+        (["Not Sure"], False, True),
+        (["Donut Devourers"], True, False),
+        (["Not Sure", "Donut Devourers"], True, True),
+    ],
+)
+def test_filter_by_form_filter_unit(
+    mockdata, units, has_officers_with_unit, has_officers_with_no_unit
+):
+    form_data = dict(unit=units)
+    unit_id = Unit.query.filter_by(descrip="Donut Devourers").one().id
+    department_id = Department.query.first().id
+
+    officers = filter_by_form(form_data, Officer.query, department_id).all()
+
+    for officer in officers:
+        found = False
+        if has_officers_with_unit:
+            found = found or any([a.unit_id == unit_id for a in officer.assignments])
+        if has_officers_with_no_unit:
+            found = found or any([a.unit_id is None for a in officer.assignments])
+        assert found
```

Comment:
Hmm, these checks make me a little concerned - in the case where both are `True`, the second conditional is totally ignored I think. Is that intended? Should we maybe set `found = True` at the beginning, then do `found = found and ...` (or something similar) so that the conditions must be met?

Reply:
When we select 2 units (ex: "Donut Devourers" and "Bureau of Organized Crime") we are looking for any officers who have at least one assignment with either the "Donut Devourerers" or "Bureau of Organized Crime" unit.

This test is checking that same case but with "Donut Devourers" and "Not Sure". It loops through the list of `officers` and asserts that each officer either `has_officers_with_unit` or `has_officers_with_no_unit`. So the second conditional would be called for any officers who weren't "Donut Devourers"

Reply:
Ahh got it! Yea I guess we only need to check that a value is returned 👍🏼